src/main/java/edu/kit/ipd/crowdcontrol/objectservice/database/operations/AbstractOperations.java

```diff
@@ -30,24 +30,29 @@ protected AbstractOperations(DSLContext create) {
     }
 
     /**
-     * executes the function if the experiment is not running.
+     * executes the function if the experiment is in draft-mode.
      * @param experimentID the id of the experiment
      * @param function the function to execute
      * @param <R> the return type
      * @return the result of the function
      * @throws IllegalStateException if the experiment is running
      */
-    protected <R> R doIfNotRunning(int experimentID, Function<Configuration, R> function) throws IllegalStateException {
+    protected <R> R doIfDraft(int experimentID, Function<Configuration, R> function) throws IllegalStateException {
         return create.transactionResult(trans -> {
-            boolean running = DSL.using(trans).fetchExists(
-                    DSL.selectFrom(Tables.TASK)
-                            .where(Tables.TASK.EXPERIMENT.eq(experimentID))
-                            .and(Tables.TASK.STATUS.eq(TaskStatus.running).or(Tables.TASK.STATUS.eq(TaskStatus.stopping)))
+            boolean draft = DSL.using(trans).fetchExists(
+                    DSL.selectFrom(Tables.EXPERIMENTS_PLATFORM_STATUS)
+                            .where(Tables.EXPERIMENTS_PLATFORM_STATUS.PLATFORM.in(
+                                    DSL.select(Tables.EXPERIMENTS_PLATFORM.IDEXPERIMENTS_PLATFORMS)
+                                            .from(Tables.EXPERIMENTS_PLATFORM)
+                                            .where(Tables.EXPERIMENTS_PLATFORM.EXPERIMENT.eq(experimentID))
+                            ))
+                            .and(Tables.EXPERIMENTS_PLATFORM_STATUS.PLATFORM_STATUS
+                                    .notEqual(ExperimentsPlatformStatusPlatformStatus.draft))
             );
-            if (!running) {
+            if (!draft) {
```

Comment:
i mean one could do that, but i think its not worth it since the method is not that complicated